spring-xd-dirt/src/main/java/org/springframework/xd/dirt/modules/metadata/FtpSourceOptionsMetadata.java

```diff
@@ -0,0 +1,170 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.modules.metadata;
+
+import org.hibernate.validator.constraints.NotBlank;
+import org.springframework.xd.module.options.mixins.FtpConnectionMixin;
+import org.springframework.xd.module.options.spi.Mixin;
+import org.springframework.xd.module.options.spi.ModuleOption;
+import org.springframework.xd.module.options.spi.ProfileNamesProvider;
+
+import javax.validation.constraints.Min;
+
+/**
+ * Module options for FTP source module.
+ *
+ * @author Franck MARCHAND
+ */
+@Mixin({FtpConnectionMixin.class})
+public class FtpSourceOptionsMetadata implements ProfileNamesProvider{
+
+    private static final String USE_REF = "use-ref";
+
+    private static final String USE_CONTENT = "use-contents";
+
+    private int clientMode = 0;
+
+    private String remoteDir = "/";
+
+    private boolean deleteRemoteFiles = false;
+
+    private String localDir = "/tmp/xd/ftp";
+
+    private boolean autoCreateLocalDir = true;
+
+    private String tmpFileSuffix = ".tmp";
+
+    private int fixedRate = 1000;
+
+    private String filenamePattern = "*";
+
+    private String remoteFileSeparator = "/";
+
+    private boolean preserveTimestamp = true;
+
+    private boolean ref = false;
+
+    @NotBlank
+    public String getRemoteDir() {
+        return remoteDir;
+    }
+
+    @ModuleOption("the remote directory to transfer the files from")
+    public void setRemoteDir(String remoteDir) {
+        this.remoteDir = remoteDir;
+    }
+
+    public boolean isDeleteRemoteFiles() {
+        return deleteRemoteFiles;
+    }
+
+    @ModuleOption("delete remote files after transfer")
+    public void setDeleteRemoteFiles(boolean deleteRemoteFiles) {
+        this.deleteRemoteFiles = deleteRemoteFiles;
+    }
+
+    @NotBlank
+    public String getLocalDir() {
+        return localDir;
+    }
+
+    @ModuleOption("set the local directory the remote files are transferred to")
+    public void setLocalDir(String localDir) {
+        this.localDir = localDir;
+    }
+
+    public boolean isAutoCreateLocalDir() {
+        return autoCreateLocalDir;
+    }
+
+    @ModuleOption("local directory must be auto created if it does not exist")
+    public void setAutoCreateLocalDir(boolean autoCreateLocalDir) {
+        this.autoCreateLocalDir = autoCreateLocalDir;
+    }
+
+    public String getTmpFileSuffix() {
+        return tmpFileSuffix;
+    }
+
+    @ModuleOption("extension to use when downloading files")
+    public void setTmpFileSuffix(String tmpFileSuffix) {
+        this.tmpFileSuffix = tmpFileSuffix;
+    }
+
+    @Min(0)
+    public int getFixedRate() {
+        return fixedRate;
+    }
+
+    @ModuleOption("fixed delay in SECONDS to poll the remote directory")
```

Comment:
I think fixed-rate for polling consumer is in milliseconds by default. isn't it ?